db/compaction/compaction_service_job.cc

```diff
@@ -475,6 +477,159 @@ static std::unordered_map<std::string, OptionTypeInfo> cs_input_type_info = {
       OptionType::kUInt64T, OptionVerificationType::kNormal,
       OptionTypeFlags::kNone}},
 };
+static std::unordered_map<std::string, OptionTypeInfo>
```

Comment:
Seems like this would be better in another file, as it could be applicable outside of compaction_service_job.cc and should be closer to where you would modify things when adding a new "built-in" table property.

Reply:
Hm.. Actually, I tried moving it to a different file and then learned that I may run into static initialization order issue. I may need a bit more time to think through. Will address this as a follow up along with the `OptionsSettableTest`